From 416b351fcdfb2c844756ed51fb7a7e483cca9c8c Mon Sep 17 00:00:00 2001
From: JYH <yahan.jiang@luxshare-ict.com>
Date: Wed, 7 Sep 2022 20:47:12 +0800
Subject: [PATCH] modify PowerPulseMs time 200 to 1000

---
 src/power_control.cpp | 19 ++++++++++++++++---
 1 file changed, 16 insertions(+), 3 deletions(-)

diff --git a/src/power_control.cpp b/src/power_control.cpp
index 05aa45c..cc36f30 100644
--- a/src/power_control.cpp
+++ b/src/power_control.cpp
@@ -130,7 +130,7 @@ static bool nmiButtonMasked = false;
 
 // This map contains all timer values that are to be read from json config
 boost::container::flat_map<std::string, int> TimerMap = {
-    {"PowerPulseMs", 200},
+    {"PowerPulseMs", 1000},
     {"ForceOffPulseMs", 15000},
     {"ResetPulseMs", 500},
     {"PowerCycleMs", 5000},
-- 
2.17.1

