From a6a800253811543124936ebae3843c1c178c3f6f Mon Sep 17 00:00:00 2001
From: IHAD <Cunjin.Tao@luxshare-ict.com>
Date: Sun, 27 Nov 2022 16:51:00 -0800
Subject: [PATCH 1/1] Change the TMP468 flag

---
 src/TMP468Main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/TMP468Main.cpp b/src/TMP468Main.cpp
index 9a34722..e1be9ff 100644
--- a/src/TMP468Main.cpp
+++ b/src/TMP468Main.cpp
@@ -51,7 +51,7 @@ static constexpr double minValueTemperature = -128; // DegreesC
 namespace fs = std::filesystem;
 static auto sensorTypes{
     std::to_array<const char*>({
-                                "xyz.openbmc_project.Configuration.TMP468Temp",
+                                "xyz.openbmc_project.Configuration.TMP468",
                                 })};
 
 static struct SensorParams
-- 
2.17.1

